Kickstarter-iOS/Features/LoginTout/Controller/LoginToutViewController.swift

```diff
@@ -187,12 +187,29 @@ public final class LoginToutViewController: UIViewController, MFMailComposeViewC
         self?.pushSignupViewController()
       }
 
-    self.viewModel.outputs.logIntoEnvironment
+    self.viewModel.outputs.logIntoEnvironmentWithApple
       .observeValues { [weak self] accessTokenEnv in
         AppEnvironment.login(accessTokenEnv)
         self?.viewModel.inputs.environmentLoggedIn()
       }
 
+    self.viewModel.outputs.logIntoEnvironmentWithFacebook
+      .observeValues { [weak self] accessTokenEnv in
+        guard let _self = self else { return }
+
+        AppEnvironment.login(accessTokenEnv)
+
+        if featureFacebookLoginDeprecationEnabled() {
+          let user = accessTokenEnv.user
+
+          user.needsPassword == true || user.needsPassword == nil
+            ? _self.pushSetYourPasswordViewController()
+            : _self.viewModel.inputs.environmentLoggedIn()
+        } else {
+          _self.viewModel.inputs.environmentLoggedIn()
+        }
```

Comment:
So this bit can be a little more readable now that `needsPassword` is non-optional.
```
let allowSetNewPassword = featureFeacbookLoginDeprecationEnabled() && accessTokenEnv.user.needsPassword

if allowSetNewPassword {
  self.pushSetYourPasswordViewController

  return
}

self.viewModel.inputs.environmentLoggedIn()
```